ci-operator/step-registry/azure/provision/bastionhost/azure-provision-bastionhost-commands.sh

```diff
@@ -181,20 +189,21 @@ vhd_blob_url=$(az storage blob url --account-name ${sa_name} --account-key ${acc
 echo "Deploy the bastion image from bastion vhd"
 run_command "az image create --resource-group ${bastion_rg} --name '${bastion_name}-image' --source ${vhd_blob_url} --os-type Linux --storage-sku Standard_LRS" || exit 2
 bastion_image_id=$(az image show --resource-group ${bastion_rg} --name "${bastion_name}-image" | jq -r '.id')
-
-echo "Create bastion subnet"
-open_port="22 873 3128 3129 5000 6001 6002" bastion_nsg="${bastion_name}-nsg" bastion_subnet="${bastion_name}Subnet"
-run_command "az network nsg create -g ${bastion_rg} -n ${bastion_nsg}" &&
-run_command "az network nsg rule create -g ${bastion_rg} --nsg-name '${bastion_nsg}' -n '${bastion_name}-allow' --priority 1000 --access Allow --source-port-ranges '*' --destination-port-ranges ${open_port}" &&
-#subnet cidr for int service is hard code, it should be a sub rang of the whole VNet cidr, and not conflicts with master subnet and worker subnet
-bastion_subnet_cidr="10.0.99.0/24"
-if [[ "${CLUSTER_TYPE}" == "azurestack" ]]; then
-    # for ash wwt, the parameter name get changed
-    vnet_subnet_address_parameter="--address-prefix ${bastion_subnet_cidr}"
-else
-    vnet_subnet_address_parameter="--address-prefixes ${bastion_subnet_cidr}"
+if [[ -z "${bastion_subnet}" ]]; then
+    echo "Create bastion subnet"
+    open_port="22 873 3128 3129 5000 6001 6002" bastion_nsg="${bastion_name}-nsg" bastion_subnet="${bastion_name}Subnet"
```

Comment:
When using master subnet provided by classic IPI install, do we need to open these port in the nsg ?

Reply:
Fixed.